### PR TITLE
[ONNX] Optional outputs debug

### PR DESCRIPTION
**Pull request by mitruska**
Degug Index Error with multioutputs, NullNode and not found blob.

### CODE REVIEW

File: inference-engine/src/inference_engine/cnn_network_ngraph_impl.cpp [hunk 1]
```diff
@@ -147,7 +147,7 @@ CNNNetworkNGraphImpl::CNNNetworkNGraphImpl(
     }
     for (auto& output : _outputData) {
         // Convert precision into native format. Be consistent with possible conversion to CNNNetwork later.
-        if (output.second->getPrecision() == Precision::I64) {
+        if (output.second->getPrecision() == Precision::I64 || output.second->getPrecision() == Precision::BOOL) {
```
Comment (mitruska) [inline inference-engine/src/inference_engine/cnn_network_ngraph_impl.cpp:150]: Without this change the tests failed with the following values error:
https://github.com/openvinotoolkit/openvino/pull/3669/checks?check_run_id=1575888630

File: ngraph/frontend/onnx_import/include/onnx_import/op/dropout.hpp [hunk 2]
```diff
@@ -39,6 +40,111 @@ namespace ngraph
...
+                        {
+                            shape_of_data = default_opset::Constant::create(
+                                ngraph::element::i32,
+                                Shape{input_data.get_partial_shape().rank().get_length()},
+                                input_data.get_partial_shape().to_shape());
+                        }
+                        else
+                        {
+                            shape_of_data = std::make_shared<default_opset::ShapeOf>(input_data);
+                        }
```
Comment (pelszkow) [inline ngraph/frontend/onnx_import/include/onnx_import/op/dropout.hpp:72]: Consider IIFE here:
```suggestion
                        std::shared_ptr<ngraph::Node> shape_of_data = [&]{
                            if (input_data.get_partial_shape().is_static())
                            {
                                return default_opset::Constant::create(
                                    ngraph::element::i32,
                                    Shape{input_data.get_partial_shape().rank().get_length()},
                                    input_data.get_partial_shape().to_shape());
                            }
                            return std::make_shared<default_opset::ShapeOf>(input_data);
                        }();
```

File: ngraph/frontend/onnx_import/include/onnx_import/op/dropout.hpp [hunk 3]
```diff
@@ -39,6 +40,111 @@ namespace ngraph
...
+                    auto inputs_size = node.get_ng_inputs().size();
+                    auto outputs_size = node.get_outputs_size();
+                    auto input_data = node.get_ng_inputs().at(0);
+
+                    if (inputs_size == 1 && outputs_size == 1)
+                    {
+                        return {input_data};
+                    }
+                    else
+                    {
```
Comment (pelszkow) [inline ngraph/frontend/onnx_import/include/onnx_import/op/dropout.hpp:56]: This `else` branch is not required.

File: ngraph/frontend/onnx_import/include/onnx_import/op/dropout.hpp [hunk 4]
```diff
@@ -39,6 +40,111 @@ namespace ngraph
...
+                            shape_of_data = std::make_shared<default_opset::ShapeOf>(input_data);
+                        }
+
+                        auto mask_value_node = default_opset::Constant::create(
+                            input_data.get_element_type(), Shape{}, {1});
+
+                        auto mask_node = std::make_shared<default_opset::Broadcast>(mask_value_node,
+                                                                                    shape_of_data);
+
+                        std::shared_ptr<default_opset::Concat> concat;
```
Comment (pelszkow) [inline ngraph/frontend/onnx_import/include/onnx_import/op/dropout.hpp:80]: Why variable is define here not inside each if?

File: ngraph/frontend/onnx_import/include/onnx_import/op/dropout.hpp [hunk 5]
```diff
@@ -39,6 +40,111 @@ namespace ngraph
...
+
+                            concat = std::make_shared<default_opset::Concat>(
+                                OutputVector{input_data, mask_node, convert_r, convert_t}, 0);
+
+                            split = std::make_shared<default_opset::Split>(
+                                concat,
+                                default_opset::Constant::create(ngraph::element::i32, Shape{}, {0}),
+                                4);
+                        }
+                        auto output_mask = std::make_shared<default_opset::Convert>(
```
Comment (pelszkow) [inline ngraph/frontend/onnx_import/include/onnx_import/op/dropout.hpp:133]: Why is not in first branch of if?

File: ngraph/frontend/onnx_import/include/onnx_import/op/dropout.hpp [hunk 6]
```diff
@@ -39,6 +40,111 @@ namespace ngraph
...
+                                broadcast_t, input_data.get_element_type());
+
+                            concat = std::make_shared<default_opset::Concat>(
+                                OutputVector{input_data, mask_node, convert_r, convert_t}, 0);
+
+                            split = std::make_shared<default_opset::Split>(
+                                concat,
+                                default_opset::Constant::create(ngraph::element::i32, Shape{}, {0}),
+                                4);
+                        }
```
Comment (pelszkow) [inline ngraph/frontend/onnx_import/include/onnx_import/op/dropout.hpp:132]: Do we need to check if `inputs_size` is bigger than 3?

**Comment by mitruska** [comment]
It was only a draft to show and discuss some examples.

File: ngraph/frontend/onnx_import/include/onnx_import/op/dropout.hpp (hunk 2 above)
Comment (mitruska) [reply ngraph/frontend/onnx_import/include/onnx_import/op/dropout.hpp:72]: Thank you for your suggestions, but it was only a draft to show and discuss some examples. Resolved with other PR.

**Pull request closed without merging**